### ISSUE TITLE
Reduce amount of incoming messages per second

### ISSUE THREAD

**Issue by nielsdeb** [joeldavidw/node-red-contrib-beacon-scanner#1]
How can I reduce the amount of messages produced by this node? I have 12 Estimote beacons in my proximity that I want to read data from every 10 seconds for example.
When I use the beacon-scanner node, it does read the data from the iBeacon and estimoteTelemetry, but it produces a massive amount of messages per second resulting in a huge clutter and CPU throttling on my Raspberry Pi B+.
I'm looking for a way to read the temperature from these 12 beacons every 10 seconds, using this node, if that's even possible?

**Comment by joeldavidw** [comment]
Hi,

Are you looking for a feature where an output will only be shown for X beacon every Y seconds? Meaning messages for duplicated beacons will only be shown every Y seconds.

**Comment by nielsdeb** [comment]
Yes, that's exactly what I'm looking for! Is there a way to achieve this using the Node you made? It's the only one capable of reading telemetry data from Estimote Beacons.

**Comment by joeldavidw** [comment]
Yep, I have just pushed the functionality to set the interval time. The default is 5000ms you can set it lower to increase the number of messages. #2

**Comment by nielsdeb** [comment]
After installing version 0.0.5 and testing a few things with it, it works exactly as I wanted! Very nice of you to add this functionality to this project, it helps me a lot! 

At the moment the new properties that you added to the node ("gracePeriod" and "duplicates") are missing for me in Node-RED. I checked your latest commit in which you did add the properties very clearly. I think this bug is related to Node-RED and has nothing to do with your node. I'll look further into this tomorrow.

**Comment by nielsdeb** [comment]
I managed to fix the issue where the node properties were missing, after that I did a lot more testing with the node and it's properties and here are my observations:
1. The "Allow Duplicates" property doesn't work correctly I think. It doesn't matter if I turn this property on or off, it never allows duplicates during a certain interval. (e.g. my beacons transmit every 200ms, in Node-RED I set a grace period of 10000ms. No matter if I allow duplicates (=true) or not (=false) I always get just 1 message for each unique beacon ID) Not sure if it supposed to work this way?
2. Each Estimote-Telemetry message consists of 2 frames. You can find an example [here](https://github.com/futomi/node-beacon-scanner#estimote-telemetry). So it might be useful to allow a duplicate Beacon ID to send a message foreach subFrameType.

With the updates you made to this node, I can do exactly what I requested. The things I mentioned in this post are just ideas to improve the node even more.

**Comment by joeldavidw** [comment]
Thank you for providing me with your feedback. 

1. I agree that the current implementation is incorrect.
2. Yep, I will update the code to allow the different subFrameType

A little caught up with exams right now. I will work on the updates later this Dec.

**Comment by joeldavidw** [comment]
Ahhh... an error on my part. I have updated the package. Do check it out and update me if its still throwing the error. Sorry I do not have a Estimote telemetry enabled beacons to test with.

**Comment by nielsdeb** [comment]
I would be glad to test out the new package for you, but I don't see the update?

**Comment by joeldavidw** [comment]
I have have pushed the update to npm. If you’re using nodered’s Palette
Manger, it might take awhile for them to receive the update.

**Comment by joeldavidw** [comment]
I have just pushed an update and tested it. It should work as expected this time round.
